### ISSUE TITLE
[14.0] traceback creating new contract  

### ISSUE THREAD

**Issue by scigghia** [OCA/contract#818]
## Module
contract

## Describe the bug
when create new contract from contract object with "recurrence at line level" and auto renew, if you change "date start" a traceback occurs

```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/opt/odoo/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
    result = request.dispatch()
  File "/opt/odoo/odoo/http.py", line 685, in dispatch
    result = self._call_function(**self.params)
  File "/opt/odoo/odoo/http.py", line 361, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/odoo/service/model.py", line 94, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/odoo/http.py", line 349, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/odoo/odoo/http.py", line 914, in __call__
    return self.method(*args, **kw)
  File "/opt/odoo/odoo/http.py", line 533, in response_wrap
    response = f(*args, **kw)
  File "/opt/odoo/addons/web/controllers/main.py", line 1394, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/opt/odoo/addons/web/controllers/main.py", line 1386, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 399, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 386, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/opt/odoo/odoo/models.py", line 6225, in onchange
    snapshot0 = Snapshot(record, nametree, fetch=(not first_call))
  File "/opt/odoo/odoo/models.py", line 6069, in __init__
    self.fetch(name)
  File "/opt/odoo/odoo/models.py", line 6079, in fetch
    self[name] = record[name]
  File "/opt/odoo/odoo/models.py", line 5686, in __getitem__
    return self._fields[key].__get__(self, type(self))
  File "/opt/odoo/odoo/fields.py", line 1021, in __get__
    self.compute_value(recs)
  File "/opt/odoo/odoo/fields.py", line 1177, in compute_value
    records._compute_field_value(self)
  File "/opt/odoo/odoo/models.py", line 4069, in _compute_field_value
    getattr(self, field.compute)()
  File "/mnt/data/odoo-addons-dir/contract/models/contract_line.py", line 134, in _compute_next_period_date_start
    super(ContractLine, rest)._compute_next_period_date_start()
  File "/mnt/data/odoo-addons-dir/contract/models/contract_recurrency_mixin.py", line 115, in _compute_next_period_date_start
    if rec.date_end and next_period_date_start > rec.date_end:
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/odoo/odoo/http.py", line 641, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/odoo/http.py", line 317, in _handle_exception
    raise exception.with_traceback(None) from new_cause
TypeError: '>' not supported between instances of 'bool' and 'datetime.date'
```

## To Reproduce
**Affected versions**: tested on 14.0 with runbot 

1. create new contract
2. set recurrence at line level
3. set auto renew
4. change date start

**Expected behavior**
Should change date start without errors

**Comment by scigghia** [comment]
![Peek 24-04-2022 15-26](https://user-images.githubusercontent.com/7657701/164978900-21af3cad-f1ec-4c37-ba2b-aafead8099a5.gif)

**Comment by pedrobaeza** [comment]
Fixed at 42b1bffef4294ba202d274f6898a1f19c1d32edf

**Comment by scigghia** [comment]
great!
